<launch>
  <!-- Arguments -->
  <arg name="config"          default="$(find unreal_airsim)/cfg/demo.yaml"/>
  <arg name="use_airsim_time" default="true"/>

  <arg name="bag_topics"      default="/airsim_drone/segmentation"/>
  <arg name="bag"             default="~/Documents/AirSim/data/airsim_segmentation.bag" />
  

  <!-- *** Run the Simulation *** -->
  
  <!-- use wsimulated time -->
  <param name="use_sim_time" value="true" if="$(arg use_airsim_time)"/>
  
  <!-- static world transform -->
  <node pkg="tf" type="static_transform_publisher" name="tf_odom_to_world" args="0 0 0 0 0 0 1 /world /odom 100"/>  

  <!-- airsim client -->
  <node name="airsim_simulator" pkg="unreal_airsim" type="airsim_simulator_node" required="true" output="screen" args="-alsologtostderr">
     <rosparam file="$(arg config)"/>
  </node>


  <!-- depth converter -->
  <node name="depth_converter" pkg="unreal_airsim" type="depth_conversion.py" required="true" output="screen"/>
  
  <!-- mesh handler-->
  <node name="mesh_handler" pkg="unreal_airsim" type="mesh_handler.py" required="true" output="screen"/>

  <!-- drone positioner -->
  <node name="drone_positioner" pkg="unreal_airsim" type="drone.py" required="true" output="screen"/>

  <!-- collector -->
  <node name="collector" pkg="unreal_airsim" type="collector.py" required="true" output="screen"/>


  <!-- Rosbag -->
  <!-- <node pkg="rosbag" type="record" name="rosbag" args="$(arg bag_topics) -O $(arg bag)" required="true"/> -->
  <!-- rosbag record /airsim_drone/segmentation -O ~/Documents/AirSim/data/airsim_segmentation.bag -->

  <!-- RVIZ Visualization -->
  <!-- <node type="rviz" name="rviz" pkg="rviz" args="-d $(find unreal_airsim)/cfg/visualization/demo.rviz"/> -->
</launch>
